src/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -1,16 +1,28 @@
 #pragma once
 
-#include <boost/coroutine2/all.hpp>
-#include <optional>
+#include <boost/coroutine2/all.hpp>  // TODO (#1888): remove this dependency
+#include <functional>
+#include <include/boost/sml.hpp>
+#include <optional>  // TODO (#1888): remove this dependency
 
-#include "software/ai/hl/stp/action/action.h"
+#include "software/ai/hl/stp/action/action.h"  // TODO (#1888): remove this dependency
 #include "software/ai/hl/stp/tactic/tactic_visitor.h"
+#include "software/ai/hl/stp/tactic/transition_conditions.h"
+#include "software/ai/intent/intent.h"
 #include "software/world/world.h"
 
+// TODO (#1888): remove this typedef
 // We typedef the coroutine return type to make it shorter, more descriptive,
 // and easier to work with
 typedef boost::coroutines2::coroutine<std::shared_ptr<Action>> ActionCoroutine;
 
+struct TacticUpdate
+{
+    Robot robot;
+    World world;
+    std::function<void(std::unique_ptr<Intent>)> set_intent;
```

Comment:
Would be nice if there was comment here on what `set_intent` should be

Reply:
done